packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -184,6 +213,84 @@ function getZeroIndexStateText(subjectNoun?: string) {
   return "No options yet";
 }
 
+function useComboboxAccessibility(
+  open: boolean,
+  selectionCallback: (selection: ComboboxOption) => void,
+  containerRef: React.RefObject<HTMLDivElement>,
+  filteredOptions: ComboboxOption[],
+  optionsListRef: React.RefObject<HTMLUListElement>,
+): void {
+  const hasOptionsVisible = open && filteredOptions.length > 0;
+  const [focusedIndex, setFocusedIndex] = useState<number | null>(null);
+
+  useEffect(() => setFocusedIndex(null), [open, filteredOptions.length]);
```

Comment:
we want to discard any previous arrow presses that might have set a focused index because with a new set of filteredOptions, there may be a new option in the first position and we wanna start fresh from `null`. same idea for when it closes. discard any index changes we made.